controller/api/models.py

```diff
@@ -110,6 +112,16 @@ def validate_domain(value):
         raise ValidationError('"{}" contains unexpected characters'.format(value))
 
 
+def validate_domain_certificate(value):
+    try:
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, value)
+        Domain.objects.get(domain=cert.get_subject().CN)
+    except crypto.Error as e:
+        raise ValidationError('Could not load certificate: {}'.format(e))
+    except Domain.DoesNotExist:
```

Comment:
I thought that heroku's ssl addon **requires** that a domain needs to be added first. That doesn't seem to be the case upon closer inspection so I'll remove the check.